packages/block-library/src/comment-template/edit.js

```diff
@@ -103,7 +103,10 @@ function CommentTemplateInnerBlocks( {
 	);
 	return (
 		<li { ...innerBlocksProps }>
-			{ comment === ( activeComment || firstComment ) ? children : null }
+			{ comment.commentId ===
```

Comment:
The comments have to be compared by `commentId` which is a string (compared by value) rather than comparing comment objects (which are compared by reference).